app/src/main/java/com/example/tasky/core/data/remote/SplashService.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.tasky.core.data.remote
+
+import okhttp3.ResponseBody
+import retrofit2.Response
+import retrofit2.http.GET
+
+interface SplashService {
+
+    @GET("/authenticate")
+    suspend fun checkAuthentication(): Response<ResponseBody>
+}
```

Comment:
Since you are checking if the user is authenticated, I would just add an endpoint in the Authentication feature, and expose a UseCase to the MainViewModel

Way less code (you don't have to create SplashService or all that logic) + it makes more sense to have a "Check if user is authenticated" on the Authentication feature